crates/uv/src/commands/pip_compile.rs

```diff
@@ -117,19 +119,21 @@ pub(crate) async fn pip_compile(
     .await?;
 
     // Check that all provided extras are used.
-    if let ExtrasSpecification::Some(extras) = extras {
-        let mut unused_extras = extras
-            .iter()
-            .filter(|extra| !used_extras.contains(extra))
-            .collect::<Vec<_>>();
-        if !unused_extras.is_empty() {
-            unused_extras.sort_unstable();
-            unused_extras.dedup();
-            let s = if unused_extras.len() == 1 { "" } else { "s" };
-            return Err(anyhow!(
-                "Requested extra{s} not found: {}",
-                unused_extras.iter().join(", ")
-            ));
+    if source_trees.is_empty() {
```

Comment:
Just wondering, why only when empty?

Reply:
It's possible that an extra _will be_ used for one of the source trees, but we can't know that until much later, since we don't have any of the metadata for the source trees yet.

Reply:
Ah okay 👍 a comment might be nice